server-common/src/main/java/org/apache/kafka/server/common/EligibleLeaderReplicasVersion.java

```diff
@@ -25,7 +25,7 @@ public enum EligibleLeaderReplicasVersion implements FeatureVersion {
     ELRV_0(0, MetadataVersion.MINIMUM_KRAFT_VERSION, Collections.emptyMap()),
 
     // Version 1 enables the ELR (KIP-966).
-    ELRV_1(1, MetadataVersion.IBP_4_0_IV1, Collections.emptyMap());
+    ELRV_1(1, MetadataVersion.IBP_4_1_IV0, Collections.emptyMap());
```

Comment:
I don't think that this will work since it will disable ELR completely for 4.0-IV1.

In any case, we need a test of ELR working with 4.0-IV1.